src/modules/dataset.ts

```diff
@@ -14,12 +14,20 @@ function updateDataset(oldVnode: VNode, vnode: VNode): void {
 
   for (key in oldDataset) {
     if (!dataset[key]) {
-      delete elm.dataset[key];
+      if (elm.dataset) {
+        delete elm.dataset[key];
+      } else {
+        elm.removeAttribute('data-' + key);
```

Comment:
less efficient on the non-critical path is just fine, I like yours better @nunocastromartins.